src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -418,7 +421,7 @@ internal void ExportCertificate(X509Certificate2 certificate, string path, bool
                             bytes = certificate.Export(X509ContentType.Pkcs12, password);
                             break;
                         case CertificateKeyExportFormat.Pem:
-                            key = certificate.GetRSAPrivateKey();
+                            key = certificate.GetRSAPrivateKey()!; // TODO - what if PEM doesn't have a private key?
```

Comment:
Are there any tests with a PEM file that doesn't have a private key?

Reply:
A private key is required to function.

Reply:
I know. At the moment I think that no private key will NRE instead of giving a useful error message.

Reply:
+ @javiercn 

We should probably throw a useful error and add a test for this.

Reply:
https://github.com/dotnet/aspnetcore/issues/28172